extensions.c

```diff
@@ -575,20 +583,29 @@ hibaext_sanity_check(const struct hibaext *ext) {
 		char *value;
 
 		while (hibaext_key_value_at(ext, i, &key, &value) == HIBA_OK) {
-			if (strcmp(key, HIBA_KEY_VALIDITY) == 0) {
+			int key_offset = 0;
+			int negative_matching = 0;
+			if (key[0] == HIBA_NEGATIVE_MATCHING)
+				negative_matching = 1;
```

Comment:
Shouldn't the key_offset be set here as well?

Reply:
Good catch